tests/infrastructure/provisioners/test_cloud_run.py

```diff
@@ -0,0 +1,338 @@
+        if command == shlex.split("gcloud --version"):
+            mock.stdout = b"Google Cloud SDK 123.456.789"
+        elif command == shlex.split("gcloud config get-value project"):
+            mock.stdout = b"test-project"
+        elif command == shlex.split("gcloud auth list --format=json"):
+            mock.stdout = json.dumps(
+                [
+                    {
+                        "account": "test-account",
+                        "status": "ACTIVE",
+                    }
+                ]
+            ).encode()
+        elif "gcloud iam service-accounts keys create" in shlex.join(command):
+            with open(command[5], "w") as f:
+                json.dump({"private_key": "test-key"}, f)
+        return mock
+
+    mock = AsyncMock(spec=run_process)
+    mock.side_effect = mock_gcloud
+    monkeypatch.setattr(
+        "prefect.infrastructure.provisioners.cloud_run.run_process", mock
+    )
+    return mock
+
+
+def assert_commands(mock: MagicMock, *commands: Union[str, re.Pattern]):
+    for i, command in enumerate(commands):
+        if isinstance(command, str):
+            assert command in shlex.join(mock.mock_calls[i].args[0])
+        else:
+            assert command.match(shlex.join(mock.mock_calls[i].args[0]))
+
+
+async def test_provision(mock_run_process, prefect_client: PrefectClient):
+    provisioner = CloudRunPushProvisioner()
+    new_base_job_template = await provisioner.provision(
+        work_pool_name="test",
+        base_job_template=default_cloud_run_push_base_job_template,
+    )
+    assert new_base_job_template
+    assert_commands(
+        mock_run_process,
+        "gcloud --version",
+        "gcloud auth list --format=json",
+        "gcloud config get-value project",
+        "gcloud config get-value run/region",
+        "gcloud services enable run.googleapis.com",
+        (
+            "gcloud iam service-accounts create prefect-cloud-run --display-name"
+            " 'Prefect Cloud Run Service Account'"
+        ),
+        (
+            "gcloud projects add-iam-policy-binding test-project"
+            " --member=serviceAccount:prefect-cloud-run@test-project.iam.gserviceaccount.com"
+            " --role=roles/iam.serviceAccountUser"
+        ),
+        (
+            "gcloud projects add-iam-policy-binding test-project"
+            " --member=serviceAccount:prefect-cloud-run@test-project.iam.gserviceaccount.com"
+            " --role=roles/run.developer"
+        ),
+        re.compile(
+            r"gcloud iam service-accounts keys create .*\/prefect-cloud-run-key\.json"
+            r" --iam-account=prefect-cloud-run@test-project\.iam\.gserviceaccount\.com"
+        ),
+    )
+
+    new_block_doc_id = new_base_job_template["variables"]["properties"]["credentials"][
+        "default"
+    ]["$ref"]["block_document_id"]
+
+    block_doc = await prefect_client.read_block_document(new_block_doc_id)
+    assert block_doc.name == "test-push-pool-credentials"
```

Comment:
We could assert that the `block_doc.data["service_account_info"]` is not `None` here maybe